tests/transforms/test_classical_jacobian.py

```diff
@@ -121,13 +129,14 @@ def test_autograd_without_argnum(circuit, args, expected_jac, diff_method):
     jac = classical_jacobian(qnode)(*args)
 
     # NOTE: We use stacking to replicate qml.jacobian behaviour for scalar-only inputs
-    if all((np.isscalar(arg) for arg in args)):
+    arg_shapes = [qml.math.shape(arg) for arg in args]
+    if len(args) == 1:
+        expected_jac = expected_jac[0]
+    if all(sh == arg_shapes[0] for sh in arg_shapes[1:]):
```

Comment:
I don't think the stacking and transpose happens for single arguments:
```suggestion
        assert np.allclose(jac, expected_jac[0])
    elif all(sh == arg_shapes[0] for sh in arg_shapes[1:]):
```

Reply:
Ah, this is to produce a proper behaviour of the _tests_, not the original code. I made no difference in the structure of `class_jacs` if there is only one argument. Therefore this unpacking is required for any test with a single argument.
The `if` clause following up is not an alternative but always to be used (note that its condition always is `True` for single arguments :) )

Reply:
Right the unpacking makes sense :) but why would you transpose the expected jacobian in the single argument case?

Reply:
Because it is the "proper" expected Jacobian that is stored in `expected_jac` (for the other interfaces) and at this point we have to mimic the weird behaviour of `qml.jacobian` when using Autograd :sweat_smile: (This is the test that I was referring to in our other discussion.)

Reply:
So the test stores the expected jacobians as tuples of individual parameter jacobians, and for autograd we need to adjust this with the stack+transpose if they are all the same shape. But I think if you look here https://github.com/PennyLaneAI/pennylane/blob/06a15ff84aac7e9991b16bf615c155fad502ce5e/pennylane/_grad.py#L216-L217, the weird bahviour (stack+transpose) exhibited by `qml.jacobian` doesn't apply to the single argument case.

Reply:
Oh, I'm sorry, you're right! I'll implement the suggestion :)

Reply:
No worries ;)